sr_world_generator/src/sr_world_generator/clean_world_file.py

```diff
@@ -0,0 +1,121 @@
+            return False
+        return True
+
+    def remove_models(self, removed_model_names: "list[str]"):
+        """
+        Removes any models with names in the supplied list from the world file.
+
+        Args:
+            removed_model_names: A list of names models that will be removed.
+        """
+        xml_root = self._xml_tree.getroot()
+        # Find all elements that have model elements as children; we need the parent reference for removal
+        for parent in xml_root.findall('.//model/..'):
+            # Find all model elements in the parent
+            for model in parent.findall('model'):
+                name = model.get('name')
+                # If the model name is in the list of models to remove, remove it
+                if name in removed_model_names:
+                    rospy.loginfo(f'Removing model: {name}')
+                    parent.remove(model)
+
+    def remove_state(self):
+        """
+        Removes the state information from the world file.
+        """
+        xml_root = self._xml_tree.getroot()
+        # Find all elements that have state elements as children; we need the parent reference for removal
+        for parent in xml_root.findall('.//state/..'):
+            for state in parent.findall('state'):
+                rospy.loginfo('Removing state information.')
+                parent.remove(state)
+
+    def save_world_file(self, output_file_path: str):
+        """
+        Saves the world file to the specified path.
+
+        Args:
+            output_file_path: The path to save the world file to.
+        """
+        # Only save the file if we're not in dry run mode
+        if not self._dry_run:
+            rospy.loginfo(f'Saving world file to {output_file_path}')
+            # Try to save the file
+            try:
+                self._xml_tree.write(output_file_path)
+            except Exception as general_exception:
+                rospy.logerr(f'Failed to save world file: {general_exception}')
+                return False
+        return True
+
+
+if __name__ == '__main__':
+    # Initialise the ROS node
+    rospy.init_node('clean_gazebo_world_file', anonymous=True)
+    # Get parameters from the ROS parameter server
+    removed_model_names_param = rospy.get_param("~removed_model_names")
+    input_file_path_param = rospy.get_param("~input_file_path")
+    output_file_path_param = rospy.get_param("~output_file_path")
+    dry_run_param = rospy.get_param("~dry_run")
+    # Create a new WorldFileCleaner object
+    world_file_cleaner = WorldFileCleaner(dry_run_param)
+    # Load the world file, remove the models and state, and save the world file
+    if not world_file_cleaner.load_world_file(input_file_path_param):
+        sys.exit(0)
+    world_file_cleaner.remove_models(removed_model_names_param)
+    world_file_cleaner.remove_state()
+    if not world_file_cleaner.save_world_file(output_file_path_param):
+        sys.exit(0)
```

Comment:
if we're expecting to have the output of this script used by something else, shouldn't it return a success if all goes well?